amd/src/editor.js

```diff
@@ -0,0 +1,993 @@
+            element.setAttribute(attribute, attributes[attribute]);
+        }
+
+        return element;
+    }
+
+    /**
+     * Utility function to check whether the current selection is inside the sup/sub tag. Returns false if it's not.
+     *
+     * @return {Boolean|ParentNode} Return the node if the selection is inside a sup/sub tag; otherwise, return false.
+     */
+    isSelectionInsideSubSup() {
+        const selection = window.getSelection();
+        if (selection.rangeCount === 0) {
+            return false;
+        }
+        const range = selection.getRangeAt(0);
+        const tagName = range.commonAncestorContainer.parentNode.nodeName;
+        // If user doesn't select any text.
+        if (selection.isCollapsed) {
+            if (this.isSupSubTag(tagName)) {
+                return range.commonAncestorContainer.parentNode;
+            }
+            return false;
+        }
+        let nodeNames;
+        const selectionNodes = range.cloneContents().childNodes;
+        for (let node of selectionNodes) {
+            const nodeName = node.nodeName;
+            if (node.textContent === '') {
+                continue;
+            }
+            if (!(this.isSupSubTag(nodeName)) &&
+                (nodeName === '#text' && !this.isSupSubTag(tagName))) {
+                return false;
+            }
+            if (!nodeNames) {
+                nodeNames = node;
+            }
+            if (!nodeNames.isEqualNode(node)) {
+                return false;
+            }
+        }
+
+        if (nodeNames.nodeName === '#text' || this.isSupSubTag(tagName)) {
+            return range.commonAncestorContainer.parentNode;
+        }
+
+        return nodeNames;
+    }
+
+    /**
+     * Check if the given tag name is 'sup' or 'sub'. Return true if it is.
+     *
+     * @param {String} tagName - Tag name need to check.
+     * @return {Boolean|ParentNode} Return the node if the selection is inside a sup/sub tag; otherwise, return false.
+     */
+    isSupSubTag(tagName) {
+        return ['SUB', 'SUP'].includes(tagName);
+    }
+
+    /**
+     * Utility function to highlight the sup/sub button.
+     *
+     * @param {String|Boolean} type - The type of the button: sup or sub.
+     */
+    setActiveButton(type) {
+        const {toolbar, button} = this.settings.classes;
+        // Deactivate all the existing buttons.
+        this.getEditor().querySelectorAll(`.${toolbar} .${button}`)
+            .forEach(button => button.classList.remove('highlight'));
+        if (type !== false) {
+            this.getSupSubButton(type)?.classList?.add('highlight');
+        }
+    }
+
+    /**
+     * Utility function to append a child node to a parent node.
+     *
+     * @param {HTMLElement} parent - The parent node that will contain the child node.
+     * @param {HTMLElement} child - The child node.
+     */
+    appendChild(parent, child) {
+        parent.appendChild(child);
+    }
+
```

Comment:
Looks like we could get rid of this function appendChild, as it's just a single line of code and we could directly call the appendChild on the node wherever its needed.